app/models/submission.rb

```diff
@@ -663,6 +941,10 @@ def submission_history
       last_submitted_at = nil
       self.versions.sort_by(&:created_at).reverse_each do |version|
         model = version.model
+        # since vericite_data is a function, make sure you are cloning the most recent vericite_data_hash
+        if self.vericiteable?
+          model.turnitin_data = self.vericite_data
```

Comment:
the "vericite_data" function has a check t o make sure it only looks up API calls for the most current version:

"self.versions.current.number == self.version_number"

it also does all the lookups in bulk and stores the data in the cache